[FIX] 풀이 응답에 공개 여부 필드 추가

jd99iam opened the pull request:
## #247 

### 💡 PR 타입(하나 이상의 PR 타입을 선택해주세요)
* [ ] 기능 추가
* [ ] 기능 삭제
* [ ] 버그 수정
* [ ] 테스트코드 작성
* [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 📲 반영 브랜치
fix/247-publicfix -> develop

### ❔ 변경 사항
- 마이페이지 풀이 응답 dto에 isPublic 추가 
- 풀이상세보기 풀이 응답 dto에 isPublic 추가
- 참고: 필드명이 is~~이라서 응답 dto에는 그냥 public으로 표시됨
- postman : 배포 후에 수정하겠음

jd99iam committed:
test(SolutionServiceTest): 테스트 수정

related to: #247

jd99iam committed:
fix(MySolutionResponse): 응답 dto 수정

풀이 응답 dto 수정
- 정적 팩터리 메서드 파라미터 수정
- isPublic 필드 추가

related to: #247

jd99iam committed:
fix(SolutionService): 정적 팩터리 메서드 호출 수정

dto의 정적 팩터리 메서드 시그니처 수정에 따른 호출 방식 수정

related to: #247

jd99iam committed:
chore(V1_init.sql): 테이블 생성 sql문 수정

테이블 생성 sql문 수정
- likes에 created_at 필드 추가
- review에 status 필드 추가

related to: #247

jd99iam committed:
fix(SolutionResponse): dto 수정

풀이 응답 dto에 isPublic 필드 추가

related to: #247

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/solution/response/MySolutionResponse.java:
```diff
@@ -16,14 +18,15 @@ public class MySolutionResponse {
-                .description(description)
-                .code(code)
+                .title(solution.getTitle())
+                .tags(Arrays.asList(solution.getAlgorithm(), solution.getDataStructure()))
```
이러면 만약에 algorithm이나 data structure가 null이면 null 값이 들어가게 되지 않나요 ?
그리고 `dataStructure`와 algorithm`을 각각 필드로 분리하지 않고 tag로 통합한 이유는 뭔가요 ?

jd99iam replied:
`    private String dataStructure;
    private String description;`
필드를 사용하도록 변경했습니다. 그리고 `get~View()` 메서드를 사용하도록 변경했습니다.
tag로 통일한 이유는 정확히 기억나지는  않지만 전에 이런 식으로 달라고 했던거 같아요..

jd99iam committed:
refactor(MySolutionResponse): dto 수정

내 풀이 응답 dto 수정
- tags를 algorithm과 datastructure 필드로 수정
- get~View() 형태 메서드 사용하도록 변경

related to: #247
